### ISSUE TITLE
positional record can't accept Attribute.

### ISSUE THREAD

**Issue by moh-hassan** [dotnet/csharplang#3644]
Records with attributes can be declared as
```cs

record Point 
{
    [Option]
    public int X { get; }
    [Option]
    public int Y { get; }
}
```
Nominal record can be declared as:

```
record Point(int X, int Y);
```

Trying to add attribute, like
```cs
record Point([Option]int X, [Option]int Y); //invalid syntax
```
Compiler raise an exception:
>CS0592 Attribute 'Option' is not valid on this declaration type. It is only valid on 'property, indexer' declarations.

As I see from this [doc](https://github.com/dotnet/csharplang/issues/3226), there is no limitation on the nominal record.
I'm using Net 5 Preview 5.
How to define attributes for the nominal record?

**Comment by YairHalberstadt** [comment]
Perhaps there should be a `[property: Option]` syntax similar to the existing `[field: Option]` syntax for auto properties?

**Comment by moh-hassan** [comment]
@YairHalberstadt ,
How your syntax match the proposal(now it is implemented) of Nominal records.

**Comment by 333fred** [comment]
@moh-hassan, you seem to be confusing nominal and positional records. This syntax is a nominal record:
```cs
record Point 
{
    [Option]
    public int X { get; }
    [Option]
    public int Y { get; }
}
```

This is a positional record:
```cs
record Point(int X, int Y);
```

Yair is proposing an additional attribute scope, presumably something like
```cs
record Point([property: Option]int X, [property: Option]int Y);
```
for the positional case.

**Comment by moh-hassan** [comment]
Thanks @333fred  for clarification.

**Comment by RikkiGibson** [comment]
See dotnet/roslyn#44691. I believe the expectation is that attributes will go on the parameter by default but if a `property:` target is specified we will put them on the synthesized property instead. This will require a diagnostic in the case that no property is synthesized for the positional parameter (like when the property is explicitly declared in the record body).

**Comment by moh-hassan** [comment]
Net 5 version [5.0.100-rc.1.20452.10](https://dotnet.microsoft.com/download/dotnet/5.0) is now supporting positional record with attributes as described by @YairHalberstadt 
```cs
public record Test(
    [field: A]
    [property: B]
    [param: C]
    [D]     
    [property: Option]
    int P1
    )
{
}
```

**Comment by 333fred** [comment]
@moh-hassan is there anything else you were requesting? Or can this be closed?

**Comment by moh-hassan** [comment]
@333fred ,
Just, I want to do more checks for the feature using Net5 Rc1.
Although VScode IDE didn't show any compilation warning,  I was surprised to get a compilation warning after cleaning for the field and property attributes, but passed for `[param ] and [D]`.
I'll feed back when test is completed.

**Comment by moh-hassan** [comment]
I can confirm that attributes for positional record are implemented in [ 5.0.100-rc.1.20452.10](https://dotnet.microsoft.com/download/dotnet/5.0). No compilation warning in VScode and I can access the attributes at runtime:
```
The attributes for the member 'Int32 P1' are:
The type of the attribute is 'B'.
The type of the attribute is 'CommandLine.OptionAttribute'.

```
@333fred 
Thanks,
You can close this issue.
